lib/master.js

```diff
@@ -246,7 +246,7 @@ Master.prototype._startWorkers = function(remainingWorkers, res) {
                 // Let all the exit listeners fire before reassigning current worker ID
                 process.nextTick(function() {
                     self._currentStartingWorker = undefined;
-                    resolve(self._startWorkers(remainingWorkers));
+                    resolve(self._startWorkers(remainingWorkers, res));
```

Comment:
If any workers succeed though (and thus decrement remaining workers) you'll lose their return values.